instrumentation/active_support/test/opentelemetry/instrumentation/active_support/span_subscriber_test.rb

```diff
@@ -205,123 +200,238 @@ def finish(name, id, payload)
-      OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name)
+    describe 'when subscribing to events matching a regular expression' do
+      let(:notification_pattern) { /.*\.foo/ }
 
-      expect do
+      it 'does not trace an event by default' do
+        ActiveSupport::Notifications.subscribe(notification_pattern) do
+          # pass
+        end
         ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
-      end.must_raise RuntimeError
+        _(last_span).must_be_nil
+      end
 
-      _(last_span).wont_be_nil
-      _(last_span.name).must_equal(notification_name)
-      _(last_span.attributes['extra']).must_equal('context')
-    end
+      it 'traces an event when a span subscriber is used' do
+        OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_pattern)
+        ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
 
-    it 'supports unsubscribe' do
-      obj = OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_name)
-      ActiveSupport::Notifications.unsubscribe(obj)
+        _(last_span).wont_be_nil
+        _(last_span.name).must_equal(notification_name)
+        _(last_span.attributes['extra']).must_equal('context')
+        _(last_span.kind).must_equal(:internal)
+      end
 
-      ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+      describe 'when using a custom span name formatter' do
+        describe 'when using the LEGACY_NAME_FORMATTER' do
+          let(:span_name_formatter) { OpenTelemetry::Instrumentation::ActiveSupport::LEGACY_NAME_FORMATTER }
+          it 'uses the user supplied formatter' do
+            OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_pattern, nil, nil, span_name_formatter: span_name_formatter)
+            ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
+
+            _(last_span).wont_be_nil
+            _(last_span.name).must_equal('foo bar')
+            _(last_span.attributes['extra']).must_equal('context')
+          end
+        end
 
-      _(obj.class).must_equal(ActiveSupport::Notifications::Fanout::Subscribers::Evented)
-      _(last_span).must_be_nil
-    end
+        describe 'when using a custom formatter' do
+          let(:span_name_formatter) { ->(name) { "custom.#{name}" } }
 
-    it 'supports setting the span kind' do
-      OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, 'bar.foo', nil, [], kind: :client)
-      ActiveSupport::Notifications.instrument('bar.foo', extra: 'context')
+          it 'uses the user supplied formatter' do
+            OpenTelemetry::Instrumentation::ActiveSupport.subscribe(tracer, notification_pattern, nil, nil, span_name_formatter: span_name_formatter)
+            ActiveSupport::Notifications.instrument(notification_name, extra: 'context')
```

Comment:
Please confirm that the indentation changes are using spaces and not tabs. There are a few other sections where it looks like indentation changed and I want to make sure that you are using spaces there as well.

Reply:
I've double checked it here and it's all spaces. :+1: